### ISSUE TITLE
Enhancement: IIS App Pools

### ISSUE THREAD

**Issue by simonwillcock** [martinlindhe/wmi_exporter#69]
It would be nice to expose IIS App Pool metrics that are available from `Win32_PerfRawData_APPPOOLCountersProvider_APPPOOLWAS` 

I think [this project's collector](https://github.com/bosun-monitor/bosun/blob/master/cmd/scollector/collectors/iis_windows.go#L115) would be a pretty useful starting point, but I've never used Go or WMI before until looking into this, so I can't say for sure.

I will try and give this a go if I can get a build environment working for Go, but someone may or may not be able to add this quicker than I might. 

On a related note, my original reason for looking into app pool metrics is that my main desire is to have the app pool as a label for sites exposed as metrics by the IIS portion of this exporter, but that's far beyond me I think. Is that data available at all? 

Thanks for your hard work on this project. The metrics it already provides are great!

**Comment by carlpett** [comment]
Hi! 
Sure, that would be a good addition, especially AppPool recycles and worker process failures can be interesting to monitor. 
Regarding labeling the sites, I'm not sure I see the use-case? A site can only have a single AppPool, right, so it wouldn't be a useful dimension to split on? It is not exposed by the IIS counters we currently look at, I think.

**Comment by simonwillcock** [comment]
Yeah the use case I was envisaging was being able to query stats for all sites in a single app pool - eg 
something like this in prometheus: 
**exposed metrics:**
```
wmi_iis_current_connections{site="www.google.com",pool="search"} 30
wmi_iis_current_connections{site="www.bing.com",pool="search"} 5
wmi_iis_current_connections{site="www.github.com",pool="git"} 30
wmi_iis_current_connections{site="www.gitlab.com",pool="git"} 10
```
**query:**
`wmi_iis_current_connections{pool="search"}`

**Comment by simonwillcock** [comment]
I'm having a crack at adding the App Pool counters in the meantime. I just found the collector-generator.. amazing! Would you suggest that I add these to the existing IIS generator? Or keep them separate? 

I'm thinking it would probably make sense to combine them?

**Comment by carlpett** [comment]
Ah, I see your point, and that would indeed be good for when you share app pools over several sites. I did a bit of reading, and I couldn't find any mapping between sites and app pools in WMI so far, but if it is there, it would be good.

I think it would be a good idea to keep them in the IIS collector.

**Comment by simonwillcock** [comment]
I've had a try at making a pull request for this - #70 

I'm not sure if this is the best way to go about combining them, but hopefully you can take it from there and just do any necessary tidying. I've tested this on my server and it appears to work well!

As for the pool label on sites, yeah I haven't found anything either yet, unfortunately. You can get the merge the two data sets manually doing things like [this](http://stackoverflow.com/questions/25088914/powershellhow-do-i-enumerate-iis-websites-and-find-the-app-pool-for-each):
```powershell
[Void][Reflection.Assembly]::LoadWithPartialName("Microsoft.Web.Administration")

$sm = New-Object Microsoft.Web.Administration.ServerManager

foreach($site in $sm.Sites)
{
    $root = $site.Applications | where { $_.Path -eq "/" }
    Write-Output ("Site: " + $site.Name + " | Pool: " + $root.ApplicationPoolName)
}
```
But I have no idea how I would even start that in Go 😄

**Comment by carlpett** [comment]
Closed via #70
